server/service/global_policies.go

```diff
@@ -164,13 +173,33 @@ func (svc Service) DeleteGlobalPolicies(ctx context.Context, ids []uint) ([]uint
 	if len(ids) == 0 {
 		return nil, nil
 	}
+	// Is there a better approach to gather policy names so we can include them in the activity records?
+	policies, err := svc.ds.ListGlobalPolicies(ctx)
+	if err != nil {
+		return nil, err
+	}
+	policiesByID := make(map[uint]*fleet.Policy, len(policies))
+	for _, p := range policies {
+		policiesByID[p.ID] = p
+	}
 	if err := svc.removeGlobalPoliciesFromWebhookConfig(ctx, ids); err != nil {
 		return nil, ctxerr.Wrap(ctx, err, "removing global policies from webhook config")
 	}
-	ids, err := svc.ds.DeleteGlobalPolicies(ctx, ids)
+	ids, err = svc.ds.DeleteGlobalPolicies(ctx, ids)
 	if err != nil {
 		return nil, err
 	}
+	// Is there a better approach to handling errors that might occur as we loop over multiple ids?
```

Comment:
Not that I can think of. Have you considered the possibility of creating one activity mentioning all the different policies instead of many activities?

Reply:
Ah, looks like `activities.details` is a `json` so it's pretty flexible. 

The UI could expect a JSON list of policy IDs+Names when the type is `fleet.ActivityTypeDeletedPolicy`.